Feature/iso timestamps

SebastianSchildt opened the pull request:
This changes timestamps to ISO8601 format as mandated by VISS V2. Fixes #160 

It should have no effect on our existing tools. but please verify Node RED examples not influenced by this

Not to code: Attaching the mandated UTC timezone "Z" is a bit ugly,  but the to_iso_extended_string method doesn't provide any options, and using the "offical" boost way of using "dateFaces" and "TimeFaces" would be many times the amount of code (and involving stringstream), so I assumed this is cleaner. I considered doing it in C, that is also more code and maybe not "nice" in a C++ Coebase


Get&Set testcase suites have a know a timestamp validation helper.
The older VssCommandProcessorTests.cpp does not include this and has been "manually fixed. Opinions? If there is a preference to also use the helper, I can change it

wenwenchenbosch commented on src/JsonResponses.cpp:
```diff
@@ -130,9 +131,24 @@ namespace JsonResponses {
+  std::string getTimeStamp(){
+    using namespace boost::posix_time;
+    ptime t = microsec_clock::universal_time();
+    return to_iso_extended_string(t)+"Z";
```
I am not a fan to use boost library if any implementation already exists in std lib. Std lib is hardware independent and more stable. Better like this?:
```
  auto now = std::chrono::system_clock::now();
  auto itt = std::chrono::system_clock::to_time_t(now);
  std::ostringstream ss;
  ss << std::put_time(gmtime(&itt), "%FT%T.%sZ");
```
https://techoverflow.net/2018/03/30/iso8601-utc-time-as-stdstring-using-c11-chrono/

wenwenchenbosch commented on test/unit-test/Gen2SetTests.cpp:
```diff
@@ -80,6 +82,15 @@ struct TestSuiteFixture {
 };
 }  // namespace
 
+//Verifies a timestamp exists and is of type string. Copies it to the
```
I read your long comment:
```
The older VssCommandProcessorTests.cpp does not include this and has been "manually fixed. Opinions? If there is a preference to also use the helper, I can change it
```
I think the preference is there, it would be nice to change it, otherwise it may be forgot. But an utils file is not necessary for now I think. More important thing for me is using std lib instead of boost.

SebastianSchildt commented:
- replaced boost based timestamp generation with function using standard APIs
- Fixed "0" default timestamp to ISO8601 format
- created helper functions to verify timestamps in unittest
- migrated more unit tests to use the new - more thorough - timestamp check

Currently CI does not build, as the boost download server for ARM builds is down

wenwenchenbosch reviewed (approved):
looks good. but jenkins can not build. pls merge after jenkin is green
